code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -473,35 +474,47 @@
 
 /obj/effect/timestop/proc/timestop()
 	playsound(get_turf(src), 'sound/magic/TIMEPARADOX2.ogg', 100, 1, -1)
-	while(loc)
-		if(duration)
-			for(var/mob/living/M in orange (freezerange, src.loc))
-				if(M in immune)
-					continue
-				M.stunned = 10
-				M.anchored = 1
-				if(istype(M, /mob/living/simple_animal/hostile))
-					var/mob/living/simple_animal/hostile/H = M
-					H.AIStatus = AI_OFF
-					H.LoseTarget()
-					continue
-			for(var/obj/item/projectile/P in orange (freezerange, src.loc))
-				P.paused = TRUE
-			duration --
-		else
-			for(var/mob/living/M in orange (freezerange+2, src.loc)) //longer range incase they lag out of it or something
-				M.stunned = 0
-				M.anchored = 0
-				if(istype(M, /mob/living/simple_animal/hostile))
-					var/mob/living/simple_animal/hostile/H = M
-					H.AIStatus = initial(H.AIStatus)
-					continue
-			for(var/obj/item/projectile/P in orange(freezerange+2, src.loc))
-				P.paused = FALSE
-			qdel(src)
-			return
+	for(var/i in 1 to duration-1)
```

Comment:
The issue isn't the `in blah to blah` part, it's that he's iterating over `duration` and then sleeping, instead of checking against `world.time` and then breaking if it fails. He's assuming that this loop has negligible run time, which is probably true in this case but not a good assumption in general. To be fair, it probably works like this, I just don't like it.